instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/message_attributes.rb

```diff
@@ -29,7 +29,8 @@ def self.set(carrier, key, value)
       #   OpenTelemetry.propagation.extract(message, getter: MessageAttributeGetter)
       class MessageAttributeGetter
         def self.get(carrier, key)
-          carrier[key][:string_value] if carrier[key][:data_type] == 'String'
+          message_attribute = carrier[key]
```

Comment:
Looks good to me, just a little nit.
Would `return unless carrier[key]` introduce less code change?

Reply:
it is less code change but since it is doing `carrier[key]` multiple times, it seems to be a little slower:

```ruby
def get_with_variable(carrier, key)
  message_attribute = carrier[key]
  message_attribute[:string_value] if message_attribute && message_attribute[:data_type] == 'String'
end

def get_with_return(carrier, key)
  return unless carrier[key]

  carrier[key][:string_value] if carrier[key][:data_type] == 'String'
end

carrier = { 'traceparent' => { data_type: 'String', string_value: 'tp' } }

Benchmark.ips do |x|
  x.report("get_with_variable") { get_with_variable(carrier, 'traceparent') }
  x.report("get_with_return") { get_with_return(carrier, 'traceparent') }

  x.compare!
end
```
```
Calculating -------------------------------------
   get_with_variable      4.959M (± 7.8%) i/s -     24.754M in   5.053304s
     get_with_return      4.337M (± 2.0%) i/s -     21.870M in   5.045064s

Comparison:
   get_with_variable:  4959117.6 i/s
     get_with_return:  4336852.7 i/s - 1.14x  slower
```

but either way is fine. let me know if I should change the code

Reply:
Thanks for putting up the comparison. I am ok with better performance.